gammapy/datasets/evaluator.py

```diff
@@ -200,7 +200,11 @@ def update(self, exposure, psf, edisp, geom, mask):
             del self._edisp_diagonal
 
         # lookup psf
-        if psf and self.model.spatial_model:
+        if (
+            psf
+            and self.model.spatial_model
+            and not (isinstance(self.psf, PSFKernel) and psf.psf_map.single_spatial_bin)
```

Comment:
Maybe one does not need the `is_single_spatial_bin` property on `Map` but directly on `IRFMap`?